.github/workflows/testing-and-coverage.yml

```diff
@@ -36,3 +36,26 @@ jobs:
       uses: codecov/codecov-action@v5
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
+
+  test-lowest-versions:
```

Comment:
Lemme convince myself: we don't need this workflow in the nightly smoke test because the smoke test is intended to catch issues with the latest versions of the dependencies creating problems. 

We should only see failures when we are submitting new code changes, as the pinned versions of the dependencies should not be changing and introducing new regressions against unchanged versions of our own code.

Reply:
I'm also not 100% convinced about that, and I don't have any more arguments for or against adding a smoke test for that.